File: DropDownList/MultiSelectDropDown/MultiSelectDropDownCSharp/CustomListVisualItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Telerik.WinControls.UI;
using System.Windows.Forms;
using System.Drawing;
using Telerik.WinControls.Enumerations;

namespace MultiSelectDropDown
{
    public class CustomListVisualItem : RadListVisualItem
    {
        private RadCheckBoxElement checkbox;
        private LightVisualElement content;

        protected override void CreateChildElements()
        {
            base.CreateChildElements();

            StackLayoutElement stack = new StackLayoutElement();
            stack.Orientation = Orientation.Horizontal;
            this.Children.Add(stack);

            checkbox = new RadCheckBoxElement();
            checkbox.ToggleStateChanged += new StateChangedEventHandler(checkbox_ToggleStateChanged);
            stack.Children.Add(checkbox);

            content = new LightVisualElement();
            content.StretchHorizontally = false;
            content.StretchVertically = true;
            content.TextAlignment = ContentAlignment.MiddleLeft;
            content.NotifyParentOnMouseInput = true;
            stack.Children.Add(content);
        }

        void checkbox_ToggleStateChanged(object sender, StateChangedEventArgs e)
        {
            ((CustomListDataItem)this.Data).Checked = this.checkbox.Checked;

            DropDownPopupForm form = this.ElementTree.Control as DropDownPopupForm;
            ((CustomEditorElement)form.OwnerDropDownListElement).SynchronizeText();
        }

        protected override Type ThemeEffectiveType
        {
            get
            {
                return typeof(RadListVisualItem);
            }
        }

        protected override void SynchronizeProperties()
        {
            base.SynchronizeProperties();
            checkbox.IsChecked = this.Data.Selected;
            this.content.Text = this.Data.Text;
            this.Text = "";
        }
    }
}

File: DropDownList/MultiSelectDropDown/MultiSelectDropDownVB/CustomListVisualItem.vb
Imports Telerik.WinControls.UI

Public Class CustomListVisualItem
    Inherits RadListVisualItem
    Private checkbox As RadCheckBoxElement
    Private content As LightVisualElement

    Protected Overrides Sub CreateChildElements()
        MyBase.CreateChildElements()

        Dim stack As New StackLayoutElement()
        stack.Orientation = Orientation.Horizontal
        Me.Children.Add(stack)

        checkbox = New RadCheckBoxElement()
        AddHandler checkbox.ToggleStateChanged, AddressOf checkbox_ToggleStateChanged
        stack.Children.Add(checkbox)

        content = New LightVisualElement()
        content.StretchHorizontally = False
        content.StretchVertically = True
        content.TextAlignment = ContentAlignment.MiddleLeft
        content.NotifyParentOnMouseInput = True
        stack.Children.Add(content)
    End Sub

    Private Sub checkbox_ToggleStateChanged(sender As Object, e As StateChangedEventArgs)
        DirectCast(Me.Data, CustomListDataItem).Checked = Me.checkbox.Checked

        Dim form As DropDownPopupForm = TryCast(Me.ElementTree.Control, DropDownPopupForm)
        DirectCast(form.OwnerDropDownListElement, CustomEditorElement).SynchronizeText()
    End Sub

    Protected Overrides ReadOnly Property ThemeEffectiveType() As Type
        Get
            Return GetType(RadListVisualItem)
        End Get
    End Property

    Protected Overrides Sub SynchronizeProperties()
        MyBase.SynchronizeProperties()
        checkbox.IsChecked = Me.Data.Selected
        Me.content.Text = Me.Data.Text
        Me.Text = ""
    End Sub
End Class